applications/nucleo_io/targets/freertos.armv6m.st-stm32f091rc-nucleo-dev-board/ServoConsumer.hxx

```diff
@@ -0,0 +1,135 @@
+#ifndef _APPLICATIONS_IO_BOARD_TARGET_SERVOCONSUMER_HXX_
+#define _APPLICATIONS_IO_BOARD_TARGET_SERVOCONSUMER_HXX_
+
+#include "config.hxx"
```

Comment:
Three reasons:
- it behaves more like a cxx file in that certain global constants it is actually initializing. This was a tradeoff that was made so that application developers would not have to carry around a config.hxx and a config.cxx file.
- For the compile CDI binary config.hxx needs to be compiled for the host which would be difficult to achieve correctly in the makefile with .o files lying around for different targets. Keeping it header-only allows compile_cdi to be created as a single compilation unit.
- Anything that includes config.hxx can by definition only live in the application tree (because that's where config.hxx exists) so it's not very portable. This makes the need for including config.hxx very low.

Reply:
I will move ServoConsumer and its ServoConsumerConfig to src/openlcb/. Because the PWM object takes in ticks (not time... why?), ServoConsumer uses configCPU_CLOCK_HZ to compute the ticks. Is this freertos-only (defined in FreeRTOSConfig.h)? What about ServoConsumer's deps on DummyGPIO and PWM.hxx, which are also freertos-specific?

For the specific problem at hand, would it be best to pass configCPU_CLOCK_HZ as a constructor parameter to ServoConsumer?

Reply:
yes that sounds like a good idea

Reply:
in `HwInit.cxx`, I see

```
Stm32PWMGroup servo_timer(TIM3,
    /*prescaler=*/ (configCPU_CLOCK_HZ * 6 / 1000 + 65535) / 65536,
    /*period_counts=*/ configCPU_CLOCK_HZ * 6 / 1000);
```

does that mean the period, as currently configured, is 6ms? shouldn't the period be 20ms for a standard servo?